ovs/extensions/support/agent.py

```diff
@@ -195,16 +200,42 @@ def _set_client(self):
         """
         if self._client is None:
             try:
-                self._client = SSHClient(endpoint='127.0.0.1')
+                self._client = SSHClient(endpoint='127.0.0.1', username='root')
             except Exception:
                 self.logger.exception('Could not instantiate a local client')
         return self._client
 
+    @staticmethod
+    def _stringify_looseversion(item):
+        if isinstance(item, LooseVersion):
+            return str(item)
+        return item
+
     def _get_package_information(self):
-        versions_dict = collections.OrderedDict()
-        for pkg_name, version in self._package_manager.get_installed_versions(client=self._client).iteritems():
-            versions_dict[pkg_name] = str(version)
-        return versions_dict
+        versions_dict = {self._client.ip: {}}
+        # ALba is always installed with OpenvStorage. The current split however offloads retrieving Alba information to the AlbaNode which is not
+        # present for non openvstorage-hc installs. Therefore explicititely request the alba information like this (otherwise update will get jeopardized).
+        final_dict = {}
+        threads = []
+
+        for fct in Toolbox.fetch_hooks(component='update', sub_component='get_package_info_cluster'):
+            thread = Thread(target=fct, args=(self._client, versions_dict))
+            thread.start()
+            threads.append(thread)
+
+        for thread in threads:
+            thread.join()
+
+        for versions in versions_dict[self._client.ip].itervalues():
+            for package, version in versions.iteritems():
+                if package in final_dict:
+                    if version != final_dict[package]:
+                        final_dict[package] = min(version, final_dict[package])
+                else:
+                    final_dict[package] = version
+        sorted_dict = sorted(final_dict.items(), key=lambda t: t[0])
```

Comment:
This is not a sorted dict but a list of tuples

Reply:
Not necessary too (can be done in your dict comprehension instead)
OrderedDict((key, self._stringify_looseversion(value)) for key, value in sorted(dict.items(), key=lamba v: v[0])